firebase-vscode/src/cli.ts

```diff
@@ -32,116 +28,75 @@ import * as commandUtils from "../../src/emulator/commandUtils";
 import { currentUser } from "./core/user";
 import { firstWhere } from "./utils/signal";
 export { Emulators };
-/**
- * Try to get a service account by calling requireAuth() without
- * providing any account info.
- */
-async function getServiceAccount() {
-  let email = null;
-  try {
-    // Make sure no user/token is sent
-    // to requireAuth which would prevent autoAuth() from being reached.
-    // We do need to send isVSCE to prevent project selection popup
-    const optionsMinusUser = await getCommandOptions(undefined, {
-      ...currentOptions.value,
-    });
-    delete optionsMinusUser.user;
-    delete optionsMinusUser.tokens;
-    delete optionsMinusUser.token;
-    email = (await requireAuth(optionsMinusUser)) || null;
-  } catch (e) {
-    let errorMessage = e.message;
-    if (e.original?.message) {
-      errorMessage += ` (original: ${e.original.message})`;
-    }
-    pluginLogger.debug(
-      `No service account found (this may be normal), ` +
-        `requireAuth error output: ${errorMessage}`,
-    );
-    return null;
-  }
-  if (process.env.WORKSPACE_SERVICE_ACCOUNT_EMAIL) {
-    // If Monospace, get service account email using env variable as
-    // the metadata server doesn't currently return the credentials
-    // for the workspace service account. Remove when Monospace is
-    // updated to return credentials through the metadata server.
-    pluginLogger.debug(
-      `Using WORKSPACE_SERVICE_ACCOUNT_EMAIL env ` +
-        `variable to get service account email: ` +
-        `${process.env.WORKSPACE_SERVICE_ACCOUNT_EMAIL}`,
-    );
-    return process.env.WORKSPACE_SERVICE_ACCOUNT_EMAIL;
-  }
-  pluginLogger.debug(
-    `Got service account email through credentials:` + ` ${email}`,
-  );
-  return email;
-}
 
 /**
  * Wrap the CLI's requireAuth() which is normally run before every command
  * requiring user to be logged in. The CLI automatically supplies it with
  * account info if found in configstore so we need to fill that part in.
+ *
  */
-async function requireAuthWrapper(showError: boolean = true): Promise<boolean> {
-  // Try to get global default from configstore. For some reason this is
+export async function requireAuthWrapper(
+  showError: boolean = true,
+): Promise<User | null> {
+  // Try to get global default from configstore
   pluginLogger.debug("requireAuthWrapper");
   let account = getGlobalDefaultAccount();
   // often overwritten when restarting the extension.
+  const accounts = getAllAccounts();
+
+  // helper to determine if VSCode options has the same account as global default
+  function isUserMatching(account: Account, options: Options) {
+
+    if (!options.user || !options.tokens) {
+      return false;
+    }
+
+    const optionsUser = options.user as User;
+    const optionsTokens = options.tokens as Tokens;
+    return (
+      account &&
+      account.user.email === optionsUser.email &&
+      account.tokens.access_token === optionsTokens.access_token
```

Comment:
Do we want to compare access_token here? Is email sufficient to determine whether this is from the right account?